src/misc/ansi_code_markup.cpp

```diff
@@ -154,6 +269,64 @@ static std::regex markup(R"((\\)?)" // Escape tag? (1)
                          ,
                          std::regex::icase);
 
+static char ansi_code[10] = {};
+
+/*!
+ * \brief Returns a pointer to an ANSI code
+ *
+ * The returned string remains valid until the next
+ * call to this function.
+ *
+ * nullptr will be returned if the provided tag
+ * is not valid.
+ *
+ * \param tag a valid tag
+ * \return const char*
+ */
+static const char *get_ansi_code(const Tag &tag)
+{
+	if (!tag.valid()) {
+		return nullptr;
+	}
+	int ansi_num = -1;
+	reset_str(ansi_code);
+	auto &tag_info = tag.info();
+	switch (tag_info.group) {
```

Comment:
"info" is a quite a generic name; when I started reading this section, I found myself losing the plot of what `info` meant.  It feels like `info()` might be exposing more of the back-end data structures - perhaps some accessor functions could hide that in this user-side code, we could directly check the colors, extents, and actions/formatting styles for a given tag?

Maybe some accessor functions in the Tag class could help avoid this?

this is minor - and I can work my way through; but if you see a way to provide some more direction comparators so the "user-side" code gets leaner.

Reply:
I've never been a huge fan of getters/setters everywhere, but I'll have another think about this, because I get where you're coming from.